search/search_service/proxy/es_search_proxy.py

```diff
@@ -0,0 +1,313 @@
+TERM_QUERY = 'term'
+TERMS_QUERY = 'terms'
+
+
+class Resource(Enum):
+    TABLE = 0
+    DASHBOARD = 1
+    FEATURE = 2
+    USER = 3
+
+
+RESOURCE_STR_MAPPING = {
+    'table': Resource.TABLE,
+    'dashboard': Resource.DASHBOARD,
+    'feature': Resource.FEATURE,
+    'user': Resource.USER,
+}
+
+
+class Filter():
+    def __init__(self, name: str, values: list, operation: str) -> None:
+        self.name = name
+        self.values = values
+        self.operation = operation
+
+
+class ElasticsearchProxy():
+    PRIMARY_ENTITIES = [Resource.TABLE, Resource.DASHBOARD, Resource.FEATURE, Resource.USER]
+
+    # mapping to translate request for table resources
+    TABLE_MAPPING = {
+        'key': 'key',
+        'badges': 'badges',
+        'tag': 'tags',
+        'schema': 'schema.raw',
+        'table': 'name.raw',
+        'column': 'column_names.raw',
+        'database': 'database.raw',
+        'cluster': 'cluster.raw'
+    }
+
+    # mapping to translate request for dashboard resources
+    DASHBOARD_MAPPING = {
+        'group_name': 'group_name.raw',
+        'name': 'name.raw',
+        'product': 'product',
+        'tag': 'tags',
+    }
+
+    # mapping to translate request for feature resources
+    FEATURE_MAPPING = {
+        'key': 'key',
+        'feature_group': 'feature_group.raw',
+        'feature_name': 'feature_name.raw',
+        'entity': 'entity',
+        'status': 'status',
+        'version': 'version',
+        'availability': 'availability.raw',
+        'tags': 'tags',
+        'badges': 'badges'
+    }
+
+    USER_MAPPING = {
+        'full_name': 'full_name',
+        'first_name': 'first_name',
+        'last_name': 'last_name',
+        'email': 'email'
+    }
+
+    RESOUCE_TO_MAPPING = {
+        Resource.TABLE: TABLE_MAPPING,
+        Resource.DASHBOARD: DASHBOARD_MAPPING,
+        Resource.FEATURE: FEATURE_MAPPING,
+        Resource.USER: USER_MAPPING,
+    }
+
+    def __init__(self, *,
+                 host: str = None,
+                 user: str = '',
+                 password: str = '',
+                 client: Elasticsearch = None) -> None:
+        # TODO actually implement this
+        if client:
+            self.elasticsearch = client
+        else:
+            http_auth = (user, password) if user else None
+            # doesn't this go against the whole point oh having a singleton pattern?
+            self.elasticsearch = Elasticsearch(host, http_auth=http_auth)
+
+
+    def _build_term_query(self, resource: Resource, query_term: str) -> Q:
+        """
+        Builds the query object for the inputed search term
+        """
+        if query_term == "" or not query_term:
+            # We don't want to create multi_match query for ""
+            # because it will result in no matches even with filters
+            return None
+
+        fields = []
+
+        # TODO make resources into an enum
+        if resource == Resource.TABLE:
+            fields = ["display_name^1000",
```

Comment:
can we have properties to store those field lists with weights? It'd be easier to fine tune them if required by certain users

Reply:
Yes, but a bit later, when we're cleaning up mappings. We know this is ugly right now.

Reply:
but the MR is supposed to improve on what we have :P just saying it doesn't seem like major refactor and it'd be in line with what we have with _MAPPING constants

Reply:
The thing is that having cleaner more general mappings will inevitably change how we handle this functionality altogether so we didn't want to spend too much time cleaning this up since it is going to change almost completely anyway soon, definitely agree that this is a part of the code that requires major improvement